File: 비밀지도.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

string decoding(int size, int n) {
	int two = pow(2, size - 1);
	string result = "";
	for (int i = 0; i < size; i++) {
		if (n >= two) {
			result += '#';
			n -= two;
		}
		else result += ' ';

		two /= 2;
	}

	return result;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
	vector<string> answer;
	for (int i = 0; i < n; i++) {
		string s1 = decoding(n, arr1[i]);
		string s2 = decoding(n, arr2[i]);

		string s3 = "";
		for (int j = 0; j < n; j++) {
			if (s1[j] == '#' || s2[j] == '#')
				s3 += '#';
			else s3 += ' ';
		}

		answer.push_back(s3);
	}

	return answer;
}

File: 비밀지도.py
def decoding(size, n):
    two = 2 ** (size-1)
    result = ""
    for i in range(size):
        if n >= two:
            result += '#'
            n -= two
        else: result += ' '
        
        two //= 2
        
    return result

def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        s1 = decoding(n, arr1[i])
        s2 = decoding(n, arr2[i])
        
        s3 = ""
        for j in range(n):
            if s1[j] == '#' or s2[j] == '#':
                s3 += '#'
            else: s3 += ' '
            
        answer.append(s3)
        
    return answer
